source/utilities.cc

```diff
@@ -745,3 +745,65 @@ namespace aspect
     template std_cxx1x::array<double,3> spherical_coordinates<3>(const Point<3> &position);
   }
 }
+
+namespace dealii
+{
+  namespace Utilities
+  {
+    namespace MPI
+    {
+      template<int dim>
+      Tensor<1,dim> 
+      reduce_tensor(const Tensor<1,dim> &local, const MPI_Comm &mpi_communicator)
+      {
+        std::vector<double> entries(dim);
+        std::vector<double> global_entries(dim);
+        Tensor<1,dim> global;
```

Comment:
Interesting, I did not know that, but it makes sense. I suppose that is part of why top-of-the-block variable declarations were widely ignored in C89?